libs/api/shared/src/lib/models/base-document.model.ts

```diff
@@ -1,10 +1,13 @@
 import { AutoMap } from '@automapper/classes';
 import { Prop } from '@nestjs/mongoose';
-import { Document } from 'mongoose';
+import { Document, Types } from 'mongoose';
 
 import { BaseModel } from './base-entity.model';
 
-export class BaseDocument extends Document implements BaseModel {
+export class BaseDocument implements BaseModel, Pick<Document, '_id'> {
+	@AutoMap()
```

Comment:
the automap seems unnecessary, as we explicitly map in the profile, right?

Reply:
Correct